### PR TITLE
Feat/deep merge config optimizations

### PR DESCRIPTION
**Pull request by tyrauber**
# Security Hardening and Performance Optimization for Style Filtering

## Summary

This PR hardens the CSS variable filtering implementation with critical security fixes and performance optimizations based on red team review.

### Commit 1: `3f1d395` - Refactor: harden style filtering with security and performance fixes

**Critical fixes:**
- Add null safety checks to prevent crashes
- Use hasOwnProperty to prevent prototype pollution attacks
- Add depth limit (100) to prevent stack overflow on deep nesting
- Fix falsy value filtering (preserve 0, false, empty strings)
- Use Object.keys() to properly filter Symbol properties
- Replace spread operator with reduce() for large array performance

**Performance improvements:**
- Extract hasNonOverlappingProperties() helper (DRY)
- Single-pass array filtering in filterCssVariables()
- Early exit in flattenStyleArray()
- Handle arrays with 10k+ items without stack overflow

**Security hardening:**
- Prevent prototype pollution via for...in loops
- Filter inherited properties with hasOwnProperty checks
- Remove Symbol properties for React Native compatibility

### Commit 2: `dffa268` - Test: add comprehensive edge case tests for security hardening

**Red team validation tests covering:**
- Circular references with depth limit protection
- Sparse arrays with undefined holes
- Objects with only Symbol properties (RN filtering)
- Mixed null/undefined/falsy values preservation
- Frozen/sealed objects handling
- Very deep nesting (105 levels) beyond limit
- Prototype pollution prevention (__proto__)
- Arrays with custom properties
- Empty arrays and objects
- Numeric string keys

**All 10 edge case tests pass, validating:**
- No crashes on malformed input
- Prototype pollution prevention
- Circular reference handling
- Stack overflow prevention
- Symbol filtering for React Native compatibility
- Null safety across all code paths

## Test Results

```
✅ 970 passing tests (42 rightIsInline-specific)
✅ 0 TypeScript errors
✅ 0 linting warnings
```

## Files Changed

- `src/native/styles/index.ts`: +66, -48
- `src/__tests__/native/rightIsInline.test.tsx`: +175

**Total:** 2 files changed, +241 insertions(+), -48 deletions(-)

### CODE REVIEW

File: src/native/styles/index.ts [hunk 1]
```diff
@@ -17,6 +17,110 @@ import {
...
+          typeof filteredItem === "object" &&
+          filteredItem !== null &&
+          Object.prototype.hasOwnProperty.call(filteredItem, VAR_SYMBOL)
+        )
+      ) {
+        filtered.push(filteredItem);
+      }
+    }
+
+    return filtered.length > 0 ? filtered : undefined;
```
[the remark this answers is not in the document]
Comment (tyrauber) [reply src/native/styles/index.ts]: The interaction between `filterCssVariables` and `flattenStyleArray` is intentional:

1. `filterCssVariables` returns `undefined` for empty arrays (after filtering)
2. Empty arrays are filtered out before reaching `flattenStyleArray`
3. `flattenStyleArray` assumes input has been pre-filtered
4. If an empty array somehow reaches `flattenStyleArray`, it's treated as non-flattenable (correct fallback)

**Comment by danstepanov** [comment]
Seems like overkill but I'm with it
